Async test cleanup

Allcharles opened the pull request:
# Implement async/await tests

Cleanup `setTimeout` and `tick()` based tests. This was replaced by the ability to create a promise which resolves once the observables `next()` or `error()` function is called.

## Changes

- Created `nStepsObservable` function which returns a Promise which will resolve after n `testStepIntervals` occur
- Created `assertImage` function to verify image `src` and `alt`
- Updated old tests to use new method
- Re-wrote the following component tests:
  - `HomeComponent`
  - `UserBadgeComponent`
  - `AppComponent`

Allcharles committed:
Created nStepObservable function

Allcharles committed:
Cleanup tests

atruskie commented on src/app/test/helpers/general.ts:
```diff
@@ -0,0 +1,30 @@
+ */
+export function nStepObservable<T>(
+  subject: Subject<T>,
+  output: T | ApiErrorDetails,
```
this should probably be a callback (or at least support an optional callback, or immediate value), otherwise the value that is returned after the async queue will have been computed *before* the method was even called.

atruskie commented on src/app/test/helpers/general.ts:
```diff
@@ -0,0 +1,30 @@
+/**
+ * Default step size for timeouts
+ */
+export const testStepInterval = 16;
```
Could this be 0?

If you're wondering how to do n steps (where n > 0) then you do multiple `setTimeouts`, one for each `steps`

atruskie replied:
If my suggestion (or assertion does not work, then leave a comment explaining why you settled on 16)

Allcharles replied:
So each step triggers the next step recursively?

Ie.

```typescript
setTimeout(() => {
  // step 0
  setTimeout(() => {
    // step 1
  }. 0);
}, 0);
```

atruskie replied:
yeah recursive, though more like:

```
var waitOne = setTimout(() => {
  if (stepsRemaining >0) {
    stepsRemaining--;
   waitOne(); 
  }
}, nextRender);
```

Allcharles committed:
Change nStepObservable to accept callback

Allcharles committed:
Change testStepInterval to 0ms

Allcharles committed:
E2E tests (#242)

- Removed french translations from angular.json
- Moved e2e/tsconfig.json to ./tsconfig.e2e.json
- Re-enabled E2E tests
- Migrated from Azure Pipelines to Github Actions
- Updated app dependencies
- Re-added favicon.ico (required for E2E tests)
